### ISSUE TITLE
"The given ColumnMapping does not match up with any column in the source or destination."

### ISSUE THREAD

**Issue by Trapulo** [borisdj/EFCore.BulkExtensions#499]
Hi, 
I have the error "The given ColumnMapping does not match up with any column in the source or destination." during a BulkInsertAsync and really I can't understand why :(

my entity is really simple:
public int ID { get; set; }
public byte[] Timestamp { get; set; }
public String EmailAddress { get; set; }
public DateTimeOffset CreationTimestamp { get; set; }

an insert using standard EF works well. Using the batchInsert I have that error but I can't understand why. The SQL table has same properties with same case... 

any idea?
thanks

**Comment by borisdj** [comment]
I have just made a test and it passes.
Code is following:
```C#
public class MyEnt
{
  public int ID { get; set; }

  [Timestamp]
  public byte[] Timestamp { get; set; }
	
  public String EmailAddress { get; set; }
	
  public DateTimeOffset CreationTimestamp { get; set; }
}
```
```C#
var entityList = new List<MyEnt>()
{
  new MyEnt { EmailAddress = "mail-1", CreationTimestamp = DateTimeOffset.Now },
  new MyEnt { EmailAddress = "mail-2", CreationTimestamp = DateTimeOffset.Now },
};
context.BulkInsert(entityList);
```

If you can write your example where it can be reproduced.

**Comment by Trapulo** [comment]
I've also some decorator:
```

 [Key]
        public int ID { get; set; }
[Required]
        public String EmailAddress { get; set; }

        [Required]
        public DateTimeOffset CreationTimestamp { get; set; }

```
I'm trying to extract a minimum code. Meantime I found that the problem is only when I try to set output identity (I will need to add child entities later in transaction):

```
  await _dbContext.BulkInsertAsync<Model.Recipient>(items, options =>
                    {
                        options.SetOutputIdentity = true;
                        options.PreserveInsertOrder = true;
                    });

```
without options it works. 

thanks

**Comment by borisdj** [comment]
I have found it, bug was on Async version and I've tested only NonAsync.
Fixed with 3.5.0.

**Comment by Trapulo** [comment]
Yes, it works now!

thanks
